### PR TITLE
Store thesaurus using persistence trait

### PR DESCRIPTION
**Pull request by mre**

### CODE REVIEW

File: crates/persistence/src/thesaurus.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    async fn load(&mut self, key: &str) -> Result<Self> {
+        let op = &self.load_config().await?.1;
+        let obj = self.load_from_operator(key, op).await?;
+        Ok(obj)
+    }
+
+    /// returns ulid as key + .json
+    fn get_key(&self) -> String {
+        // TODO: What should the key be?
+        "thesaurus.json".to_string()
```
Comment (mre) [inline crates/persistence/src/thesaurus.rs:36]: @AlexMikhalev, any idea for what we could use as a key here?

Comment (AlexMikhalev) [reply crates/persistence/src/thesaurus.rs:36]: "role shortname"+"thesaurus.json".to_string()
there is always a default role in a config

Comment (mre) [reply crates/persistence/src/thesaurus.rs:36]: We don't have the roles in the `Thesaurus`.

Comment (mre) [reply crates/persistence/src/thesaurus.rs:36]: I was thinking about passing the role to the constructor, like `Thesaurus::new("role")`, but the `new` method in `Persistable` trait doesn't take any arguments.

Comment (AlexMikhalev) [reply crates/persistence/src/thesaurus.rs:36]: Check out https://github.com/terraphim/terraphim-ai/blob/main/crates/config/src/lib.rs - now has a default, which can be the default prefix, as in the default role.

Comment (mre) [reply crates/persistence/src/thesaurus.rs]: Good idea. Done.

**Pull request merged**